test/SlowTests/Server/Documents/DataArchival/DataArchivalEtlTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FastTests;
+using FastTests.Utils;
+using Raven.Client;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Operations.DataArchival;
+using Raven.Client.Documents.Operations.ETL;
+using Raven.Client.Util;
+using SlowTests.Core.Utils.Entities;
+using Sparrow;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Server.Documents.DataArchival;
+
+public class DataArchivalEtlTests : RavenTestBase
+{
+    public DataArchivalEtlTests(ITestOutputHelper output) : base(output)
+    {
+
+    }
+    
+    
+    private async Task SetupDataArchival(DocumentStore store)
+    {
+        var config = new DataArchivalConfiguration {Disabled = false, ArchiveFrequencyInSec = 100};
+
+        await DataArchivalHelper.SetupDataArchival(store, Server.ServerStore, config);
+    }
+
+    [Fact]
+    public async Task CanFilterOutUnarchivedDocumentsInEtl()
+    {
+        using (var src = GetDocumentStore())
+        using (var dest = GetDocumentStore())
+        {
+            Etl.AddEtl(src, new RavenEtlConfiguration
+            {
+                Name = "simulate",
+                ConnectionStringName = "my-con",
+                Transforms = new List<Transformation>
+                {
+                    new()
+                    {
+                        Collections = {"Companies"},
+                        Name = "SkipUnarchived",
+                        Script =
+                            "var archived = this[\"@metadata\"][\"@archived\"]" + Environment.NewLine +
+                            "if (archived !== undefined && archived === true) {" + Environment.NewLine +
```

Comment:
We need to have a test which verifies that users can filter out archived documents as well

Reply:
done here 8863287fd408ae593ef5062c8ddd2f806d1382da